src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -271,9 +270,9 @@ protected TupleBatch fetchNextReady() throws IOException, DbException {
                 finishedReadingLastRow = true;
                 break;
               }
-              dataChar = trailingEndInputStream.read();
+              dataChar = partitionInputStream.read();
```

Comment:
Note that a simple optimization would be to increment a byte position count here, which we could use as the ending byte index in the constructor for `completePartitionStream`.